lib/api.js

```diff
@@ -354,11 +350,13 @@ api.declare({
   name: 'terminateInstance',
   title: 'Terminate an instance',
   stability: API.stability.experimental,
-  deferAuth: true,
-  scopes: [
-    ['ec2-manager:manage-instances:<region>:<instanceId>'],
-    ['ec2-manager:manage-resources:<workerType>'],
-  ],
+  scopes: {AnyOf: [
+    'ec2-manager:manage-instances:<region>:<instanceId>',
+    {
+      if: 'hasWorkerType',
+      then: 'ec2-manager:manage-resources:<workerType>',
```

Comment:
is this if limited to booleans or is it more expressive?

Reply:
Yeah, just limited to booleans just to keep the surface area of the expression language quite small.